interleave.py

```diff
@@ -0,0 +1,18 @@
+#submitted by Keren Or Hadad 208025205
+# part 5.4
+import itertools
+
+
+def interleave(*lists):
+    if len(set(len(_) for _ in lists)) > 1:
+        raise ValueError("Lists are not all the same length!")
+    joint = list(itertools.chain(*lists))
+    for l_idx, li in enumerate(lists):
```

Comment:
The names of the variables here should be more informative :)

Reply:
done